### PR TITLE
Add RideDetailPage tests

### PR DESCRIPTION
**Pull request by fynsta**
Second try without broken diff

### CODE REVIEW

**Comment by Showcas** [comment]
please don't forget the page when you name the test. Just thought we are doing the same tests

**Review by Showcas — APPROVED** [review]

**Comment by fynsta** [comment]
> please don't forget the page when you name the test for the PR titel. Just thought we are doing the same tests 

whoops my bad :D

File: lib/rides/models/ride.dart [hunk 1]
```diff
@@ -203,6 +203,10 @@ extension RideStatusExtension on RideStatus {
     return this == RideStatus.approved;
   }
 
+  bool isRealRider() {
```
Comment (istzen) [inline lib/rides/models/ride.dart:206]: can we add a comment, what this method checks, i think the name is kinda weird

Comment (fynsta) [reply lib/rides/models/ride.dart:206]: Yes I'm not happy with the name either... Any ideas?

File: test/widgets/ride_detail_page_test.dart [hunk 2]
```diff
@@ -0,0 +1,554 @@
...
+        ride = RideFactory().generateFake(
+          status: RideStatus.preview,
+          drive: NullableParameter(null),
+          driveId: drive.id,
+        );
+
+        when(processor.processUrl(any)).thenReturn(jsonEncode(drive.toJsonForApi()));
+      });
+
+      testWidgets('It loads the corresponding drive', (WidgetTester tester) async {
```
Comment (istzen) [inline test/widgets/ride_detail_page_test.dart:110]: i don't understand the test description

File: test/widgets/ride_detail_page_test.dart [hunk 3]
```diff
@@ -0,0 +1,554 @@
...
+    });
+
+    group('when ride is cancelled by driver', () {
+      setUp(() {
+        ride = RideFactory().generateFake(status: RideStatus.cancelledByDriver, drive: NullableParameter(drive));
+
+        when(processor.processUrl(any)).thenReturn(jsonEncode(ride.toJsonForApi()));
+      });
+
+      testWidgets('it shows the riders, a hide button and a banner', (WidgetTester tester) async {
```
Comment (istzen) [inline test/widgets/ride_detail_page_test.dart:191]: why are we showing the riders here but not when the ride is cancelledbyRider?

File: test/widgets/ride_detail_page_test.dart [hunk 4]
```diff
@@ -0,0 +1,554 @@
...
+          final Ride returnedRide = RideFactory().generateFake(status: RideStatus.pending);
+          when(processor.processUrl(any)).thenReturn(jsonEncode(returnedRide.toJsonForApi()));
+
+          final Finder requestRideYesButton = find.byKey(const Key('requestRideYesButton'));
+          expect(requestRideYesButton, findsOneWidget);
+          await tester.tap(requestRideYesButton);
+          await tester.pumpAndSettle();
+
+          // Verify that the ride was requested (but no way to verify body right now)
+          // One call to update the ride, one call to load it again
```
Comment (istzen) [inline test/widgets/ride_detail_page_test.dart:376]:
```suggestion
          // One call to insert the ride, one call to load it again
```

File: lib/rides/models/ride.dart (hunk 1 above)
Comment (istzen) [reply lib/rides/models/ride.dart:206]: no that's the problem xD, maybe just a short comment or something.

File: test/widgets/ride_detail_page_test.dart (hunk 2 above)
Comment (fynsta) [reply test/widgets/ride_detail_page_test.dart:110]: Yeah no idea what that was supposed to mean

File: test/widgets/ride_detail_page_test.dart (hunk 3 above)
Comment (fynsta) [reply test/widgets/ride_detail_page_test.dart:191]: well if its cancelledByDriver, then the state is basically frozen - no one can request or withdraw etc. So it's fine if show all the other riders (only overlapping of course) that were equally screwed by the cancellation.

But if it's cancelledByRider, then you are not a part of the drive anymore, so you should not see any riders that are added after you cancelled for example

File: lib/rides/models/ride.dart (hunk 1 above)
Comment (fynsta) [reply lib/rides/models/ride.dart:206]: maybe "canSeeFullRideInformation" ?

File: test/widgets/ride_detail_page_test.dart (hunk 3 above)
Comment (istzen) [reply test/widgets/ride_detail_page_test.dart:191]: that makes sense thanks

File: lib/rides/models/ride.dart (hunk 1 above)
Comment (istzen) [reply lib/rides/models/ride.dart:206]: yes that's good then it's clear why we need the method.

**Review by istzen — APPROVED** [review]

File: test/widgets/ride_detail_page_test.dart [hunk 5]
```diff
@@ -0,0 +1,556 @@
...
+import 'package:motis_mitfahr_app/welcome/pages/login_page.dart';
+import 'package:motis_mitfahr_app/welcome/pages/register_page.dart';
+
+import '../util/factories/drive_factory.dart';
+import '../util/factories/model_factory.dart';
+import '../util/factories/profile_factory.dart';
+import '../util/factories/review_factory.dart';
+import '../util/factories/ride_factory.dart';
+import '../util/mock_server.dart';
+import '../util/mock_server.mocks.dart';
```
Comment (Eddie-42) [inline test/widgets/ride_detail_page_test.dart:29]: Hast einen Rebase verkackt:
```
Target of URI doesn't exist: '../util/mock_server.mocks.dart'.
Try creating the file referenced by the URI, or Try using a URI for a file that does exist.
```

**Review by Eddie-42 — APPROVED** [review]

File: test/widgets/ride_detail_page_test.dart (hunk 5 above)
Comment (fynsta) [reply test/widgets/ride_detail_page_test.dart:29]: Jaja bin am fixen... ist nur leider bisschen mehr

**Pull request merged**